main/init/gradle/src/mill/main/gradle/GradleBuildGenMain.scala

```diff
@@ -0,0 +1,336 @@
+    } finally connector.disconnect()
+  }
+
+  private def writeGradleInitScript: os.Path = {
+    val file = os.temp.dir() / "init.gradle"
+    val classpath =
+      os.Path(classOf[ProjectTreePlugin].getProtectionDomain.getCodeSource.getLocation.toURI)
+    val plugin = classOf[ProjectTreePlugin].getName
+    val contents =
+      s"""initscript {
+         |    dependencies {
+         |        classpath files(${escape(classpath.toString()) /* escape for Windows */})
+         |    }
+         |}
+         |
+         |allprojects {
+         |    apply plugin: $plugin
+         |}
+         |""".stripMargin
+    os.write(file, contents)
+    file
+  }
+
+  def getBaseInfo(
+      input: Tree[Node[ProjectModel]],
+      cfg: Config,
+      baseModule: String,
+      packagesSize: Int
+  ): IrBaseInfo = {
+    val project = {
+      val projects = input.nodes(Tree.Traversal.BreadthFirst).map(_.value).toSeq
+      cfg.baseProject
+        .flatMap(name => projects.collectFirst { case m if name == m.name => m })
+        .orElse(projects.collectFirst { case m if null != m.maven().pom() => m })
+        .orElse(projects.collectFirst { case m if !m.maven().repositories().isEmpty => m })
+        .getOrElse(input.node.value)
+    }
+    if (packagesSize > 1) {
+      println(s"settings from ${project.name()} will be shared in base module")
+    }
+    val supertypes =
+      Seq("MavenModule") ++
+        Option.when(null != project.maven().pom()) { "PublishModule" }
+
+    val javacOptions = getJavacOptions(project)
+    val repos = getRepositories(project)
+    val pomSettings = extractPomSettings(project)
+    val publishVersion = getPublishVersion(project)
+
+    val typedef = IrTrait(
+      cfg.shared.jvmId,
+      baseModule,
+      supertypes,
+      javacOptions,
+      pomSettings,
+      publishVersion,
+      Nil,
+      repos
+    )
+
+    IrBaseInfo(javacOptions, repos, pomSettings == null, publishVersion, Seq.empty, typedef)
+  }
+
+  override def extractIrBuild(
+      cfg: Config,
+      baseInfo: IrBaseInfo,
+      build: Node[ProjectModel],
+      packages: Map[(String, String, String), String]
+  ): IrBuild = {
+    val project = build.value
+    val scopedDeps = extractScopedDeps(project, packages, cfg)
+    val version = getPublishVersion(project)
+    IrBuild(
+      scopedDeps = scopedDeps,
+      testModule = cfg.shared.testModule,
+      hasTest = os.exists(getMillSourcePath(project) / "src/test"),
+      dirs = build.dirs,
+      repositories = getRepositories(project).diff(baseInfo.repositories),
+      javacOptions = getJavacOptions(project).diff(baseInfo.javacOptions),
+      projectName = getArtifactId(project),
+      pomSettings = if (baseInfo.noPom) extractPomSettings(project) else null,
+      publishVersion = if (version == baseInfo.publishVersion) null else version,
+      packaging = null,
+      pomParentArtifact = null,
+      resources = Nil,
+      testResources = Nil,
+      publishProperties = Nil
```

Comment:
@ajaychandran here as well, we should either leave a comment as to why `packaging`/`pomParentArtifact`/`resources`/`testResources`/`publishProperties` are null/Nil? Or fill them out if they do not need to be so

Reply:
Added support for `packaging` / `publsihProperties`.
Added comments for 
- `pomParentArtifact`: This is not directly available in the Gradle API.
- `resources`/`testResources`: These were skipped to avoid using  `JavaPluginExtension.getSourceSets`; this would limit support to Gradle version `7.1+`.